mdsuite/visualizer/znvis_visualizer.py

```diff
@@ -0,0 +1,142 @@
+"""
+MDSuite: A Zincwarecode package.
+
+License
+-------
+This program and the accompanying materials are made available under the terms
+of the Eclipse Public License v2.0 which accompanies this distribution, and is
+available at https://www.eclipse.org/legal/epl-v20.html
+
+SPDX-License-Identifier: EPL-2.0
+
+Copyright Contributors to the Zincwarecode Project.
+
+Contact Information
+-------------------
+email: zincwarecode@gmail.com
+github: https://github.com/zincware
+web: https://zincwarecode.com/
+
+Citation
+--------
+If you use this module please cite us with:
+
+Summary
+-------
+Module implementing the ZnVis visualizer in MDSuite.
+"""
+import importlib.resources
+import json
+
+import numpy as np
+import znvis
+from PIL.ImageColor import getcolor
+
+from mdsuite.database.simulation_database import Database
+from mdsuite.utils.meta_functions import join_path
+
+
+class SimulationVisualizer:
+    """
+    Visualize a simulation.
+    """
+
+    def __init__(
+        self,
+        species: list = None,
+        unwrapped: bool = False,
+        frame_rate: float = 24,
+        database_path: str = None,
+    ):
+        """
+        Constructor for the visualizer.
+
+        Parameters
+        ----------
+        species : list
+                A list of species to visualize.
+        unwrapped : bool
+                If true, unwrapped coordinates are studied.
+        frame_rate : float
+                Frame rate at which to run the visualization in frames per second.
+        database_path : str
+                Database path from the experiment.
+        """
+        self.counter = 0
+        # Particle information
+        self.database = Database(name=join_path(database_path, "database.hdf5"))
+        self.frame_rate = frame_rate
+        self.species = species
+        if unwrapped:
+            self.identifier = "Unwrapped_Positions"
+        else:
+            self.identifier = "Positions"
+
+    @staticmethod
+    def _get_species_properties(species: str):
+        """
+        Collect species properties from pubchempy file.
+
+        Parameters
+        ----------
+        species : float
+                Species to load data for.
+
+        Returns
+        -------
+        colour : np.ndarray
+                RBG array of colours.
+        radius : float
+                Radius of the particles. This is a reduced mass.
+        """
+        # Load the species data from pubchempy data file.
+        data_name = "mdsuite.data"
+        json_name = "PubChemElements_all.json"
+        with importlib.resources.open_text(data_name, json_name) as json_file:
+            pse = json.loads(json_file.read())
```

Comment:
I'd prefer using the following, because it will support refactoring:

```py
import mdsuite.data

import importlib.resources as pkg_resources

pse = json.loads(pkg_resources.read_text(mdsuite.data, "PubChemElements_all.json"))
```

Reply:
yep that is a good point.

Reply:
resolved in 71ff782